### PR TITLE
fix: use the original class name in the class body

### LINKED ISSUE #4811: self reference in static block is not working well on rename
### Rollup Version

3.10.0

### Operating System (or Browser)

Chrome

### Node Version (if applicable)

_No response_

### Link To Reproduction

https://rollupjs.org/repl/?version=3.10.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMiUyRiolMjBUUkVFLVNIQUtJTkclMjAqJTJGJTVDbmltcG9ydCUyMCU3QiUyMFRlc3QlMjAlN0QlMjBmcm9tJTIwJy4lMkZ0ZXN0MS5qcyclM0IlNUNuaW1wb3J0JTIwJTdCJTIwVGVzdCUyMGFzJTIwVGVzdDIlMjAlN0QlMjBmcm9tJTIwJy4lMkZ0ZXN0Mi5qcyclM0IlNUNuY29uc29sZS5sb2coVGVzdC50ZXN0JTJDJTIwVGVzdDIudGVzdCklMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJ0ZXN0MS5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjIlNUNuZXhwb3J0JTIwY2xhc3MlMjBUZXN0JTIwJTdCJTVDbiUyMCUyMHN0YXRpYyUyMHRlc3QlMjAlM0QlMjAnVGVzdDEnJTNCJTVDbiUyMCUyMHN0YXRpYyUyMCU3QiUyMGNvbnNvbGUubG9nKFRlc3QudGVzdCklM0IlMjAlN0QlNUNuJTdEJTIyJTJDJTIyaXNFbnRyeSUyMiUzQWZhbHNlJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMnRlc3QyLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMiU1Q25leHBvcnQlMjBjbGFzcyUyMFRlc3QlMjAlN0IlNUNuJTIwJTIwc3RhdGljJTIwdGVzdCUyMCUzRCUyMCdUZXN0MiclM0IlNUNuJTIwJTIwc3RhdGljJTIwJTdCJTIwY29uc29sZS5sb2coVGVzdC50ZXN0KSUzQiUyMCU3RCU1Q24lN0QlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXMlMjIlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlN0QlMkMlMjJleGFtcGxlJTIyJTNBbnVsbCU3RA==

### Expected Behaviour

When running, it should give the following in console.

```
Test1
Test2
Test1 Test2
```

### Actual Behaviour

Currently it give an error:
```
VM93:3 Uncaught ReferenceError: Test$1 is not defined
    at <static_initializer> (<anonymous>:3:24)
    at <anonymous>:1:14
```

### PR DESCRIPTION
**Pull request by TrickyPi**
This PR contains:

- [x] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

List any relevant issue numbers:
resolves #4811 

### Description
The `Variable` renders a proper name according to the return of the `useOriginalName` function.

### CODE REVIEW

File: test/form/samples/use-class-name-in-static-block/_config.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+	description: 'use class name instead of renderName in static block',
+	solo: true
```
Comment (lukastaegert) [inline test/form/samples/use-class-name-in-static-block/_config.js:3]: 🫢

File: src/utils/renderHelpers.ts [hunk 2]
```diff
@@ -15,6 +15,7 @@ export interface RenderOptions {
 	namespaceToStringTag: boolean;
 	pluginDriver: PluginDriver;
 	snippets: GenerateCodeSnippets;
+	useOriginalName?: (variable: Variable) => boolean;
```
Comment (lukastaegert) [inline src/utils/renderHelpers.ts:18]: That is a really minimal solution, I like that. Very minor point: For performance reasons, it is better if objects passed to functions always have the same properties. Then the compiler can produce more optimized code. Thus, a better type would be

```js
useOriginalName: ((variable: Variable) => boolean) | null
```

and you add an explicit `null` where the RenderOptions are created.

**Review by lukastaegert — COMMENTED** [review]
Very nice!

File: src/utils/renderHelpers.ts (hunk 2 above)
Comment (TrickyPi) [reply src/utils/renderHelpers.ts:18]: > For performance reasons, it is better if objects passed to functions always have the same properties. Then the compiler can produce more optimized code

Oh, I don' know this, could you give me some cases that can demonstrate this?

File: test/form/samples/use-class-name-in-static-block/_config.js (hunk 1 above)
Comment (TrickyPi) [reply test/form/samples/use-class-name-in-static-block/_config.js:3]: Uhhh, What does this mean?

Comment (lukastaegert) [reply test/form/samples/use-class-name-in-static-block/_config.js:3]: You forgot to remove the "solo"...

Comment (TrickyPi) [reply test/form/samples/use-class-name-in-static-block/_config.js:3]: Ohhh! I forgot it.

File: src/utils/renderHelpers.ts (hunk 2 above)
Comment (lukastaegert) [reply src/utils/renderHelpers.ts:18]: Ok, it is a technical thing, but if you want to do some reading, I found this: https://erdem.pl/2019/08/v-8-function-optimization#function-states

Basically if you pass objects of different shapes to a function, that makes the function "polymorphic", which means that the generated machine code first needs to determine the shape of the object and then call the correct branch. If all objects passed to a function have the same shape, the function can be monomorphic and will be much faster.

Comment (lukastaegert) [reply src/utils/renderHelpers.ts:18]: It is admittedly a very low-level optimization, but it is also cheap to fix.

**Comment by lukastaegert** [comment]
For the failing tests, note that you can add a parameter `minNodeVersion` to `_config.js` files to skip tests on older Node versions, i.e.

```js
module.exports = {
  description: 'use class name instead of renderName in static block',
  minNodeVersion: 16
};
```

File: src/utils/renderHelpers.ts (hunk 2 above)
Comment (TrickyPi) [reply src/utils/renderHelpers.ts:18]: Ok, I get it, Thanks very much!

**Comment by TrickyPi** [comment]
> For the failing tests, note that you can add a parameter `minNodeVersion` to `_config.js` files to skip tests on older Node versions, i.e.
> 
> ```js
> module.exports = {
>   description: 'use class name instead of renderName in static block',
>   minNodeVersion: 16
> };
> ```

Ok

**Review by lukastaegert — APPROVED** [review]

**Pull request merged**